homeassistant/components/tplink/switch.py

```diff
@@ -30,57 +31,60 @@ async def async_setup_entry(
     """Set up switches."""
     data: TPLinkData = hass.data[DOMAIN][config_entry.entry_id]
     parent_coordinator = data.parent_coordinator
-    device = cast(SmartPlug, parent_coordinator.device)
-    if not device.is_plug and not device.is_strip and not device.is_dimmer:
-        return
+    device = parent_coordinator.device
+    add_device_switch = True
+    for no_switch_module in NO_SWITCH_DEVICE_MODULES:
+        if no_switch_module in device.modules:
+            add_device_switch = False
+            break
```

Comment:
This is to prevent creating switches for lights.  Will probably not be needed once https://github.com/home-assistant/core/pull/117758 is merged.

Reply:
I'd avoid adding this as it's getting replaced soon anyway.

Reply:
https://github.com/home-assistant/core/pull/117898 created to remove this in the next PR.